### PR TITLE
Pubdev 5650 newchunk timing test

### PR DESCRIPTION
**Pull request by wendycwong**
Added NewChunkSpeedTest.java to test the time taken to create integer/long/double vectors.

### CODE REVIEW

**Comment by wendycwong** [comment]
Here is some timing results with nrow = 1000000, 2 loops


  | Master Results | Spencer
-- | -- | --
  | mean | mean
BigDoubles | 0.89 | 0.9297
BigDoubleConstants | 1.441333 | 0.8417
Double | 1.127667 | 0.1873
DoubleConstant | 0.3285 | 0.31
Float | 2.1555 | 1.9193
FloatConstant | 0.321333 | 0.247
Long |   | 1.89
LongConstants |   | 0.6363
Integer | 0.798 | 0.57
IntegerConstants | 1.332667 | 1.618

File: h2o-core/src/jmh/java/water/NewChunkBench.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+public class NewChunkBench {
+
+  @Param({"10000", "1000000"})
+  private int rows;
+  private double[] rawDouble;
+  private double[] rawFloat;
+  private double[] rawLong;
+  private double[] rawInt;
+
```
Comment (michalkurka) [inline h2o-core/src/jmh/java/water/NewChunkBench.java]: please also check the performance of constant columns

File: h2o-core/src/jmh/java/water/NewChunkBench.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Chunk access patterns benchmark
+ */
+@State(Scope.Thread)
+//@Fork(value = 1, jvmArgsAppend = "-XX:+PrintCompilation")
+@Fork(value = 1, jvmArgsAppend = "-Xmx12g")
+@Warmup(iterations = 5)
```
Comment (mmalohlava) [inline h2o-core/src/jmh/java/water/NewChunkBench.java]: Do you know that 5 is enough? Meaning did you check JIT, that after 5iterations, there is no heavy jitting?

File: h2o-core/src/test/java/water/fvec/NewChunkSpeedTest.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    double endTime = (System.currentTimeMillis()-startTime)*0.001;  // change time to seconds
+    Log.info("New Chunk test for constant doubles:", " time(s) taken for "+numberLoops+" loops is "+endTime);
+  }
+
+  @Test public void testParseBigDoublesConst(){
+    double startTime = System.currentTimeMillis();
+    for (int index=0; index<numberLoops; index++)
+      testsForDoubles(true, true, false);
+    double endTime = (System.currentTimeMillis()-startTime)*0.001;  // change time to seconds
+    Log.info("New Chunk test for big constant doubles:", " time(s) taken for "+numberLoops+" loops is "+endTime);
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/fvec/NewChunkSpeedTest.java]: you can use https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/main/java/water/util/PrettyPrint.java to print time.

**Comment by mmalohlava** [comment]
btw: why are you running u-bench with 3nodes?

**Comment by wendycwong** [comment]
Michal M:

I just run it in multi-node mode with 3 nodes.  Running with more nodes can be too slow.  Just  rough timing test here.  


W

File: h2o-core/src/test/java/water/fvec/NewChunkSpeedTest.java (hunk 3 above)
Comment (wendycwong) [reply h2o-core/src/test/java/water/fvec/NewChunkSpeedTest.java]: Thanks, I have incorporated it.

File: h2o-core/src/jmh/java/water/NewChunkBench.java (hunk 2 above)
Comment (wendycwong) [reply h2o-core/src/jmh/java/water/NewChunkBench.java]: Yes, from my runs off my little laptop, the timing does not vary much.

**Pull request merged**